src/main/java/net/neoforged/neoforge/common/IOUtilities.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common;
+
+import java.io.BufferedOutputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.function.BiPredicate;
+import java.util.function.Consumer;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.NbtIo;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Declares a class containing helpers for performing file I/O in a resillient
+ * manner.
+ */
+public final class IOUtilities {
+    private static final String TEMP_FILE_SUFFIX = ".neoforge-tmp";
+    private static final OpenOption[] OPEN_OPTIONS = {
+            StandardOpenOption.DSYNC, // TODO: DSYNC or SYNC here?
```

Comment:
This is the one main question I have here: should we use DSYNC or SYNC? DSYNC only requires the content be written synchronously, while SYNC also waits for the metadata to finish.

Reply:
Oh, oops - I misclicked and resolved before posting. In any case, we discussed this in the NeoForge server and while it would be better if we could change this to use `FileChannel` to give us access to `FileChannel.force(false)`, it would be harder to implement due to needing to expose streams to consumers. Because of this, I'm going to leave it as `DSYNC` and leave changing the backend to FileChannel to somebody in the future, if performance becomes an issue.

Reply:
I think you may have forgotten to remove the comment, or at least replace it with this explanation.

Reply:
Unresolving since you forgot to fix it: